airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DefaultConfigPersistence.java

```diff
@@ -34,27 +34,37 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 // we force all interaction with disk storage to be effectively single threaded.
 public class DefaultConfigPersistence implements ConfigPersistence {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultConfigPersistence.class);
   private static final String CONFIG_DIR = "config";
 
   private static final Object lock = new Object();
 
   private final JsonSchemaValidator jsonSchemaValidator;
+  private final Path storageRootWithConfigDirectory;
   private final Path storageRoot;
 
-  public DefaultConfigPersistence(final Path storageRoot) {
-    this(storageRoot, new JsonSchemaValidator());
+  public DefaultConfigPersistence(final Path storageRootConfig) {
```

Comment:
You shouldn't rename this argument since it still has the same name in the other ctor.

Reply:
and actually calling it `storageRootConfig` makes it sound that it contains the config_dir suffix